# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: toolkit/components/thumbnails/PageThumbs.jsm
# Commit: a400e87a7082
# Full Hash: a400e87a708271cc1830228b4d05ff19b455136c
# Author: Emma Malysz <emalysz@mozilla.com>
# Date: 2021-05-22 03:53:20
# Description:
#   Bug 1711634, reduce number of AsyncShutdownTimeout crashes from PageThumbs r=barret
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D115407
# ==============================================================================

diff -r 978c022b695d -r a400e87a7082 toolkit/components/thumbnails/PageThumbs.jsm
--- a/toolkit/components/thumbnails/PageThumbs.jsm	Fri May 21 23:21:58 2021 +0300
+++ b/toolkit/components/thumbnails/PageThumbs.jsm	Fri May 21 20:27:16 2021 +0000
@@ -33,7 +33,6 @@
 
 XPCOMUtils.defineLazyModuleGetters(this, {
   Services: "resource://gre/modules/Services.jsm",
-  AsyncShutdown: "resource://gre/modules/AsyncShutdown.jsm",
   PageThumbUtils: "resource://gre/modules/PageThumbUtils.jsm",
   PrivateBrowsingUtils: "resource://gre/modules/PrivateBrowsingUtils.jsm",
 });
@@ -686,12 +685,12 @@
     //    which will eventually be fixed by bug 965309)
     //
 
-    let blocker = () => promise;
+    let blocker = () => undefined;
 
     // The following operation will rise an error if we have already
     // reached profileBeforeChange, in which case it is too late
     // to clear the thumbnail wipe.
-    AsyncShutdown.profileBeforeChange.addBlocker(
+    IOUtils.profileBeforeChange.addBlocker(
       "PageThumbs: removing all thumbnails",
       blocker
     );
@@ -707,12 +706,7 @@
     } finally {
       // Generally, we will be done much before profileBeforeChange,
       // so let's not hoard blockers.
-      if ("removeBlocker" in AsyncShutdown.profileBeforeChange) {
-        // `removeBlocker` was added with bug 985655. In the interest
-        // of backporting, let's degrade gracefully if `removeBlocker`
-        // doesn't exist.
-        AsyncShutdown.profileBeforeChange.removeBlocker(blocker);
-      }
+      IOUtils.profileBeforeChange.removeBlocker(blocker);
     }
   },
 